packages/Liquid-Tests.package/LQObjectRepoTests.class/instance/setUp.st

```diff
@@ -0,0 +1,4 @@
+running
```

Comment:
As I understand it you would define a superclass `LQObjectRepoTests` with two subclasses `LQLocalObjectRepoTests` and `LQRemoteObjectRepoTests`, which implement all of the tests for their respective ObjectRepo? I would second that opinion, this makes it easier to define which test uses which ObjectRepo, plus we can make sure to test all tests on both Repos.

Reply:
> As I understand it you would define a superclass `LQObjectRepoTests` with two subclasses `LQLocalObjectRepoTests` and `LQRemoteObjectRepoTests`, which implement all of the tests for their respective ObjectRepo? I would second that opinion, this makes it easier to define which test uses which ObjectRepo, plus we can make sure to test all tests on both Repos.

Yes, thats about what I meant. But is there should be no implementation difference for the tests, right? The only difference is what client they use, so the only change in the subclasses would be defining a different client in setUp

Reply:
Yes, it would be just like the discussed change of adding a superclass for all tests. Actually, should we implement that idea as well, we wouldn't even need an extra setUp for the LocalObjectRepo-Tests, since that would be done in the superclass.
How I would do it now is to implement all tests in the `ObjectRepoTests`-superclass, and to call `super testNameHere` in the subclasses with their client as argument

Reply:
@Skn0tt The messagePassing-Test only works on a RemoteRepo, correct? I don't see how we would do it with a LocalRepo

Reply:
The newest commit includes this change, @PaulVII please check if this is what you meant.

Reply:
> The newest commit includes this change, @PaulVII please check if this is what you meant.

Yes that's about it ^^ I realize now that what I actually wanted is not really possible, since there cannot be an abstract test case class - it's tests would fail. Locally, I made the remote tests inherit from the local ones, this way we could get rid of five or so more duplicate methods. If you think this would be worth it, I can push that as a commit.

<img width="711" alt="grafik" src="https://user-images.githubusercontent.com/67124476/125312786-e4b7fe00-e334-11eb-8618-1bbcd419a361.png">
<img width="690" alt="grafik" src="https://user-images.githubusercontent.com/67124476/125312816-eaaddf00-e334-11eb-9200-a7dea3787a1e.png">

Reply:
Before you do that: Note that currently, your `LQRemoteObjectRepoTests` only has one test. It does not inherit the tests, meaning it does not test the tests implemented in its superclass, if you do not specifically tell it to do so

Reply:
It seems that I made a mistake - the tests do actually get inherited, I tested something similar, but different as I just found out. So your proposal could be be a good idea - how many methods would we get rid of?

Reply:
Okay, I seem to have found out how we can _actually_ do this: 
1. If a class has tests, it does _not_ inherit the tests from its superclass.
2. If the class does not have any tests, it _does_ inherit the tests from its superclass.

Since No. 2 applies in our case, I had a look into `TestCase` and found that if the superclass is abstract, or the current class does not have any tests, it inherits the selectors of its superclass. By just overwriting the class-side method `shouldInheritSelectors`, we can force `LQRemoteObjectRepoTests` to also perform all the tests of its superclass.
When I tested this (see my last comment), I must have had a class without any tests itself, which is why it performed the tests of the superclass.
The next commit includes this changed class hierarchy, which saves us two methods for each added test performed on both Repo-Types.